components/transaction/internal/adapters/http/in/transaction-helpers.go

```
@@ -0,0 +1,122 @@
+
+package in
+
+import (
+	"github.com/LerianStudio/midaz/v3/pkg/constant"
+	"github.com/LerianStudio/midaz/v3/pkg/mmodel"
+	pkgTransaction "github.com/LerianStudio/midaz/v3/pkg/transaction"
+)
+
+// deriveTransactionAction returns the action constant based on whether
+// the transaction is pending (hold) or immediate (direct).
+func deriveTransactionAction(pending bool) string {
```

golangci-lint (2.11.3)

[error] func deriveTransactionAction is unused

(unused)